File: web/src/lib/guideContent.ts
import type { LucideIcon } from 'lucide-react';
import {
  Package,
  UserPlus,
  Truck,
  Receipt,
  FileText,
  Sparkles,
  Link2,
  MessageSquare,
  BarChart3,
  CloudUpload,
  Bell,
} from 'lucide-react';

export const GUIDE_VERSION = '2.0';

export interface GuideSection {
  icon: LucideIcon;
  iconColor: string;
  iconBg: string;
  title: string;
  body: string;
}

export const GUIDE_SECTIONS: GuideSection[] = [
  {
    icon: Package,
    iconColor: 'text-violet-700 dark:text-violet-400',
    iconBg: 'bg-violet-100 dark:bg-violet-950/50',
    title: '1. Danh mục sản phẩm',
    body: `Vào **Sản phẩm** trên sidebar trước khi ghi đơn.

Mỗi sản phẩm có: tên, **giá vốn**, **giá bán mặc định**, **hoa hồng mặc định**, ghi chú nội bộ (tuỳ chọn).

**Theo dõi tồn kho** (tuỳ chọn):
• **Bật** — hàng vật lý: nhập tồn, ngưỡng cảnh báo. Ghi đơn tự trừ kho; Bảng điều khiển báo sắp hết / hết.
• **Tắt** — dịch vụ, đặt hộ, dropship… không quản số lượng.

Danh mục liên kết khách ↔ đơn ↔ tiền ↔ kho (nếu bật).`,
  },
  {
    icon: UserPlus,
    iconColor: 'text-blue-700 dark:text-blue-400',
    iconBg: 'bg-blue-100 dark:bg-blue-950/50',
    title: '2. Khách hàng, địa chỉ & nguồn',
    body: `Nhấn **Thêm khách** hoặc **Khách hàng → Thêm khách**.

Điền:
• **Tên** (bắt buộc), SĐT, **địa chỉ giao hàng mặc định**, ghi chú
• **Nguồn khách** — Facebook, Zalo, TikTok, giới thiệu… (xem doanh thu theo kênh trong Thống kê)
• **Sản phẩm quan tâm** — chọn từ danh mục (khuyến nghị) hoặc nhập tay
• **Trạng thái** — Mới / Tiềm năng / Nóng / Đã chốt (quyết định lịch nhắc chăm)
• **Ngày hết bảo hành** (tuỳ chọn)

**Địa chỉ trên hồ sơ khách** là mặc định cho đơn mới — sửa hồ sơ **không** làm đổi địa chỉ trên đơn đã ghi (xem mục 3).`,
  },
  {
    icon: Truck,
    iconColor: 'text-cyan-700 dark:text-cyan-400',
    iconBg: 'bg-cyan-100 dark:bg-cyan-950/50',
    title: '3. Ghi đơn & giao hàng (snapshot)',
    body: `Từ **Chi tiết khách → Ghi đơn**, hoặc **Đơn hàng → Ghi đơn mới**.

Form ghi đơn có block **Thông tin giao hàng** — tự điền từ hồ sơ khách:
• **Người nhận**, **SĐT**, **Địa chỉ** — sửa được ngay lúc ghi đơn (giao hộ, địa chỉ tạm…)
• Khi lưu, Salenote **snapshot** 3 trường này vào đơn — cố định, không đổi khi bạn sửa hồ sơ khách sau này

Phần sản phẩm & tiền:
• Chọn SP từ danh mục (tự điền giá) hoặc nhập tay; cảnh báo nếu vượt tồn kho
• SL, giá bán / vốn / hoa hồng, thanh toán (đủ / cọc / nợ)
• Tuỳ chọn đánh dấu khách **Đã chốt** sau khi ghi đơn`,
  },
  {
    icon: FileText,
    iconColor: 'text-emerald-700 dark:text-emerald-400',
    iconBg: 'bg-emerald-100 dark:bg-emerald-950/50',
    title: '4. Bill, copy ship & cập nhật đơn',
    body: `Trong **Chi tiết khách** (từng dòng đơn) hoặc **Đơn hàng → mở đơn**:

• **Sao chép giao hàng** — copy block người nhận / SĐT / địa chỉ / SP / tiền dán sang shipper hoặc Zalo
• **Xem bill** — preview phiếu bán hàng → **Tải PDF** hoặc **Chia sẻ**
• Sửa **giao hàng trên đơn** (người nhận, SĐT, địa chỉ) — badge **Khác hồ sơ khách** nếu khác profile
• Cập nhật **thanh toán** (cọc / nợ) nếu đơn chưa thu đủ

**Cài đặt → Thông tin trên bill**: nhập tên shop & SĐT hiển thị trên PDF.

Danh sách **Đơn hàng** có phân trang; mỗi đơn hiển thị doanh thu và trạng thái thanh toán.`,
  },
  {
    icon: Sparkles,
    iconColor: 'text-fuchsia-700 dark:text-fuchsia-400',
    iconBg: 'bg-fuchsia-100 dark:bg-fuchsia-950/50',
    title: '5. Trợ lý Sale — Bảng điều khiển',
    body: `**Bảng điều khiển** (Trang chủ) không chỉ liệt kê khách — có **Trợ lý Sale** gợi ý việc làm hôm nay:

• **Việc nên làm hôm nay** — ưu tiên nhắn khách, thu nợ, gửi ưu đãi, chăm khách cũ…
• **Cảnh báo rủi ro** — khách nóng lâu chưa liên hệ, công nợ cao, bảo hành sắp hết
• **Gợi ý doanh thu** — so sánh tuần/tháng, top SP, kênh bán tốt
• **Thành tích** — streak liên hệ 🔥, mốc đơn / doanh thu

Bấm từng gợi ý để nhảy thẳng tới khách hoặc màn hình liên quan. **Chi tiết khách** còn có panel **Thông minh khách** (tổng đơn, doanh thu, gợi ý hành động).`,
  },
  {
    icon: Link2,
    iconColor: 'text-orange-700 dark:text-orange-400',
    iconBg: 'bg-orange-100 dark:bg-orange-950/50',
    title: '6. Liên kết dữ liệu Salenote',
    body: `Mọi thứ nối với nhau trên máy bạn:

**Khách** → nguồn + SP quan tâm + địa chỉ mặc định
**Đơn hàng** → khách + SP + snapshot giao hàng + doanh thu / lời / HH / công nợ
**Dashboard & Trợ lý Sale** → tính từ đơn và lịch sử liên hệ thực
**Thống kê** → doanh thu theo SP và **nguồn khách**

Xoá khách sẽ xoá luôn đơn và lịch sử liên hệ — cẩn thận trước khi xoá.`,
  },
  {
    icon: MessageSquare,
    iconColor: 'text-sky-700 dark:text-sky-400',
    iconBg: 'bg-sky-100 dark:bg-sky-950/50',
    title: '7. Chăm khách & nhắc lịch',
    body: `**Bảng điều khiển** hiển thị khách cần liên hệ ngay và sắp tới.

🔵 **Mới** — nhắc sau 1 ngày
🟠 **Tiềm năng** — nhắc sau 2 ngày
🔴 **Nóng** — nhắc mỗi ngày
🟢 **Đã chốt** — nhắc chăm lại sau 7 ngày

Bấm **Đã nhắn** để ghi nhận liên hệ (reset đồng hồ nhắc). Trong chi tiết khách: soạn tin mẫu, sao chép sang Zalo/Messenger, gọi 1 chạm (app). Mẫu tin tuỳ chỉnh tại **Cài đặt**.`,
  },
  {
    icon: BarChart3,
    iconColor: 'text-purple-700 dark:text-purple-400',
    iconBg: 'bg-purple-100 dark:bg-purple-950/50',
    title: '8. Thống kê',
    body: `**Thống kê** (theo tháng):
• Doanh số thực từ đơn hàng (doanh thu, lời, HH, công nợ)
• Lượt liên hệ, khách mới, streak 🔥
• Top SP bán chạy (theo doanh thu đơn)
• **Doanh thu theo nguồn khách** — biết kênh nào mang tiền

**Khách hàng**: lọc theo trạng thái và nguồn; tìm theo tên, SĐT, địa chỉ, SP.
**Chi tiết khách**: tất cả đơn + tổng doanh thu / lời / HH / nợ + thông minh khách.`,
  },
  {
    icon: CloudUpload,
    iconColor: 'text-teal-700 dark:text-teal-400',
    iconBg: 'bg-teal-100 dark:bg-teal-950/50',
    title: '9. Sao lưu dữ liệu',
    body: `Dữ liệu lưu trên máy bạn (IndexedDB web / SQLite app), không gửi lên server.

**Cài đặt → Sao lưu & khôi phục**:
• **Tải backup JSON** — khách, liên hệ, sản phẩm, đơn hàng (kèm snapshot giao hàng)
• **Khôi phục JSON** — file dùng chung giữa web và app mobile
• **Xuất CSV** — khách (kèm nguồn), liên hệ, đơn hàng

Nên backup ít nhất 1 lần/tuần.`,
  },
  {
    icon: Bell,
    iconColor: 'text-amber-700 dark:text-amber-400',
    iconBg: 'bg-amber-100 dark:bg-amber-950/50',
    title: '10. Thông báo nhắc nhở',
    body: `**Cài đặt → Thông báo nhắc nhở**:

• **Hàng ngày** — giờ tuỳ chọn: khách cần liên hệ (Mới/Tiềm năng/Nóng)
• **Tổng kết tuần** (Thứ Hai) — doanh số 7 ngày, số đơn
• **Tổng kết tháng** (ngày 1) — liên hệ & chốt đơn tháng trước
• **Tri ân & ưu đãi** (Thứ Sáu) — khách tiềm năng lâu chưa chăm, khách cũ nên gửi quà

**Web**: nhắc khi tab Salenote đang mở (hoặc PWA đã cài). **App mobile**: nhắc nền cả khi đóng app.`,
  },
];

File: app/lib/data/guide_content.dart
import 'package:flutter/material.dart';

/// Phiên bản hướng dẫn — đồng bộ web/src/lib/guideContent.ts
const String guideVersion = '2.3';

class GuideSectionData {
  final IconData icon;
  final Color iconColor;
  final String title;
  final String body;

  const GuideSectionData({
    required this.icon,
    required this.iconColor,
    required this.title,
    required this.body,
  });
}

/// Nội dung đồng bộ với web/src/lib/guideContent.ts (+ mục riêng app mobile)
List<GuideSectionData> salenoteGuideSections({bool includeMobile = true}) {
  final sections = <GuideSectionData>[
    const GuideSectionData(
      icon: Icons.inventory_2_outlined,
      iconColor: Color(0xFF6D28D9),
      title: '1. Danh mục sản phẩm',
      body:
          'Vào Sản phẩm trước khi ghi đơn.\n\n'
          'Mỗi SP có: giá vốn, giá bán mặc định, hoa hồng mặc định, ghi chú nội bộ.\n\n'
          'Theo dõi tồn kho (tuỳ chọn):\n'
          '• Bật — hàng vật lý: nhập tồn, cảnh báo sắp hết. Ghi đơn tự trừ kho.\n'
          '• Tắt — dịch vụ / đặt hộ / không quản kho.\n\n'
          'Danh mục liên kết khách ↔ đơn ↔ tiền ↔ kho.',
    ),
    const GuideSectionData(
      icon: Icons.person_add_outlined,
      iconColor: Color(0xFF1565C0),
      title: '2. Khách hàng, địa chỉ & nguồn',
      body:
          'Thêm khách từ FAB hoặc menu Khách hàng.\n\n'
          'Điền: tên, SĐT, địa chỉ giao hàng mặc định, nguồn (Facebook, Zalo…), '
          'SP từ danh mục hoặc nhập tay, trạng thái, bảo hành.\n\n'
          'Địa chỉ trên hồ sơ khách là mặc định cho đơn mới — sửa hồ sơ không đổi địa chỉ trên đơn đã ghi.',
    ),
    const GuideSectionData(
      icon: Icons.local_shipping_outlined,
      iconColor: Color(0xFF0891B2),
      title: '3. Ghi đơn & giao hàng (snapshot)',
      body:
          'Chi tiết khách → Ghi đơn, hoặc menu → Ghi đơn / Đơn hàng.\n\n'
          'Block Thông tin giao hàng tự điền từ khách — có thể sửa người nhận, SĐT, địa chỉ trước khi lưu.\n\n'
          'Salenote snapshot 3 trường này vào đơn — cố định, không đổi khi sửa hồ sơ khách sau này.\n\n'
          'Nhập SL, giá bán/vốn/HH, thanh toán (đủ / cọc / nợ). Tuỳ chọn đánh dấu Đã chốt.',
    ),
    const GuideSectionData(
      icon: Icons.receipt_long_outlined,
      iconColor: Color(0xFF047857),
      title: '4. Bill, copy ship & cập nhật đơn',
      body:
          'Chi tiết khách hoặc Đơn hàng → mở đơn:\n\n'
          '• Copy ship — sao chép thông tin giao hàng\n'
          '• Xem bill — preview → Tải PDF / Chia sẻ\n'
          '• Sửa giao hàng trên đơn (badge Khác hồ sơ nếu khác profile)\n'
          '• Cập nhật thanh toán nếu chưa thu đủ\n\n'
          'Cài đặt → Thông tin trên bill: tên shop & SĐT trên PDF.\n\n'
          'Danh sách Đơn hàng hiển thị doanh thu và trạng thái thanh toán. '
          'Trên app: cuộn xuống cuối để tải thêm (xem mục 11).',
    ),
    const GuideSectionData(
      icon: Icons.auto_awesome_outlined,
      iconColor: Color(0xFFC026D3),
      title: '5. Trợ lý — Trang Hôm nay',
      body:
          'Trang chủ là trợ lý kinh doanh cá nhân:\n\n'
          '• Việc hôm nay — nhắn khách, thu nợ, mời mua lại, nhập hàng sắp hết…\n'
          '• Mục tiêu tháng — tiến độ doanh thu (đặt trong Cài đặt)\n'
          '• Gợi ý hôm nay — kết luận từ dữ liệu tháng\n'
          '• Ai nợ tôi — danh sách khách còn nợ\n\n'
          'Bấm từng việc để nhảy tới khách hoặc màn hình liên quan. '
          'Chi tiết khách có panel trí nhớ (tổng đơn, doanh thu, gợi ý).',
    ),
    const GuideSectionData(
      icon: Icons.link_rounded,
      iconColor: Color(0xFFEA580C),
      title: '6. Liên kết dữ liệu Salenote',
      body:
          'Khách → nguồn + SP quan tâm + địa chỉ mặc định\n'
          'Đơn hàng → khách + SP + snapshot giao hàng + doanh thu / lời / HH / công nợ\n'
          'Hôm nay & Trợ lý → tính từ đơn và lịch sử liên hệ thực\n'
          'Tiền của tôi → doanh thu theo SP và nguồn khách\n\n'
          'Xoá khách sẽ xoá luôn đơn và lịch sử liên hệ.',
    ),
    const GuideSectionData(
      icon: Icons.chat_bubble_outline,
      iconColor: Color(0xFF0284C7),
      title: '7. Chăm khách & nhắc lịch',
      body:
          'Hôm nay hiển thị khách cần liên hệ (mục Sổ khách hôm nay — mở rộng khi cần).\n\n'
          '🔵 Mới — nhắc sau 1 ngày\n'
          '🟠 Tiềm năng — 2 ngày\n'
          '🔴 Nóng — mỗi ngày\n'
          '🟢 Đã chốt — chăm lại sau 7 ngày\n\n'
          'Vuốt thẻ hoặc Chi tiết → Nhắn tin / Gọi. Mẫu tin tuỳ chỉnh trong Cài đặt.',
    ),
    const GuideSectionData(
      icon: Icons.bar_chart_outlined,
      iconColor: Color(0xFF7C3AED),
      title: '8. Thống kê',
      body:
          'Thống kê: doanh số đơn thực, lượt liên hệ, khách mới, top SP bán, '
          'doanh thu theo nguồn, streak.\n\n'
          'Sổ khách: lọc trạng thái, nguồn; tìm tên, SĐT, địa chỉ, SP. '
          'Danh sách dài: cuộn xuống cuối để tải thêm (mục 11).\n\n'
          'Chi tiết khách: tất cả đơn + tổng doanh thu / lời / HH / nợ.',
    ),
    const GuideSectionData(
      icon: Icons.cloud_upload_outlined,
      iconColor: Color(0xFF0D9488),
      title: '9. Sao lưu & đồng bộ web',
      body:
          'Cài đặt → Sao lưu JSON (khách, liên hệ, SP, đơn kèm snapshot giao hàng).\n\n'
          'File JSON dùng chung với web Salenote — export app, import web và ngược lại.\n\n'
          'Nên backup mỗi tuần. PIN và cài đặt thông báo không nằm trong JSON.',
    ),
    const GuideSectionData(
      icon: Icons.view_list_outlined,
      iconColor: Color(0xFF475569),
      title: '10. Danh sách dài (cuộn tải thêm)',
      body:
          'App không tải hết hàng nghìn dòng một lúc — hiển thị 20 mục đầu, '
          'rồi tải thêm khi bạn cuộn.\n\n'
          'Áp dụng: Sổ khách, Sản phẩm, Đơn hàng, Ai nợ tôi, sổ hôm nay trên Trang chủ, '
          'form chọn khách/SP.\n\n'
          'Cuộn xuống cuối danh sách để tự tải thêm. Cuối list có dòng '
          '«Cuộn để xem thêm (x/y)» hoặc «Đã hiển thị tất cả». '
          'Không cần bấm nút chuyển trang.\n\n'
          'Nếu danh sách ngắn hơn màn hình mà vẫn còn dữ liệu, app tự tải thêm — '
          'bạn luôn xem được toàn bộ sổ.\n\n'
          'Trên web: trang danh sách đầy đủ dùng nút Phân trang; popup chọn khách/SP cuộn tải thêm giống app.',
    ),
  ];

  if (includeMobile) {
    sections.add(
      const GuideSectionData(
        icon: Icons.notifications_active_outlined,
        iconColor: Color(0xFFF57F17),
        title: '11. Riêng app mobile',
        body:
            'Cài đặt → Thông báo:\n'
            '• Hàng ngày — giờ tuỳ chọn, khách cần liên hệ\n'
            '• Tổng kết tuần (T2) — doanh số 7 ngày\n'
            '• Tổng kết tháng (ngày 1) — thống kê tháng trước\n'
            '• Tri ân & ưu đãi (T6) — khách tiềm năng & khách cũ\n\n'
            'Thêm: gọi điện 1 chạm, PIN khóa app, vuốt thẻ nhanh, nhắc nền 24/7.\n'
            'Web nhắc khi tab mở; app nhắc cả khi đóng app.',
      ),
    );
  }

  return sections;
}
